code/game/machinery/floodlight.dm

```diff
@@ -12,74 +11,76 @@
 	active_power_usage = 200
 	power_channel = LIGHT
 	use_power = POWER_USE_OFF
-	
+
 	machine_name = "emergency floodlight"
 	machine_desc = "A portable, battery-powered LED flood lamp used to illuminate large areas."
 
-	//better laser, increased brightness & power consumption
-	var/l_max_bright = 0.8 //brightness of light when on, can be negative
-	var/l_inner_range = 0 //inner range of light when on, can be negative
-	var/l_outer_range = 4.5 //outer range of light when on, can be negative
+	/// Brightness of the light when on. Can be negative.
+	var/lamp_brightness = 0.8
+	/// Inner range of the light when on. Can be negative
+	var/lamp_inner_range = 0
+	/// Outer range of the light when on. Can be negative.
+	var/lamp_outer_range = 4.5
 
 /obj/machinery/floodlight/on_update_icon()
-	icon_state = "flood[panel_open ? "o" : ""][panel_open && get_cell() ? "b" : ""]0[use_power == POWER_USE_ACTIVE]"
+	overlays.Cut()
+	// We build the floodlight's appearance using overlays based on its status
+	if (use_power == POWER_USE_ACTIVE)
+		overlays.Add("floodlight_on")
+	if (panel_open)
+		overlays.Add("floodlight_open")
+		if (get_cell())
+			overlays.Add("floodlight_open_cell")
 
 /obj/machinery/floodlight/power_change()
 	. = ..()
-	if(!. || !use_power) return
-
-	if(stat & NOPOWER)
-		turn_off(1)
+	if (!. || !use_power)
+		return
+	if (stat & NOPOWER)
+		turn_off(TRUE)
 		return
 
-	// If the cell is almost empty rarely "flicker" the light. Aesthetic only.
-	if(prob(30))
-		set_light(l_max_bright / 2, l_inner_range, l_outer_range)
-		spawn(20)
-			if(use_power)
-				set_light(l_max_bright, l_inner_range, l_outer_range)
-
-// Returns 0 on failure and 1 on success
-/obj/machinery/floodlight/proc/turn_on(var/loud = 0)
-	if(stat & NOPOWER)
-		return 0
-
-	set_light(l_max_bright, l_inner_range, l_outer_range)
+/// Turns on the floodlight, returning TRUE on a success or FALSE otherwise. If loud is defined, it will show a message and play a sound.
+/obj/machinery/floodlight/proc/turn_on(loud)
```

Comment:
Since all calls to this specify TRUE anyway, why not make it a default parameter?

```suggestion
/obj/machinery/floodlight/proc/turn_on(loud = TRUE)
```